multiplier_analysis/nbconverted/1_get_multiplier_LV_coverage.py

```diff
@@ -0,0 +1,249 @@
+# To do this in python you can use the following code:
+# from scipy.stats import norm
+# norm.ppf((0.05/17754)/2)
+# Here we are using a p-value = 0.05
+# with a Bonferroni correction for 17754 tests, which are
+# the number of P. aeruginosa genes
+
+zscore_threshold = 4.68
+ls_genes_out = process.get_generic_specific_genes(ls_data_files, zscore_threshold)
+
+
+# ## Get LV data and filter
+
+# In[6]:
+
+
+# Load multiplier models
+# Converted formatted pickle files (loaded using phenoplier environment) from
+# https://github.com/greenelab/phenoplier/blob/master/nbs/01_preprocessing/005-multiplier_recount2_models.ipynb
+# into .tsv files
+# Raw data was downloaded from https://figshare.com/articles/recount_rpkm_RData/5716033/4
+multiplier_model_summary = pd.read_csv("multiplier_model_summary.tsv", sep="\t", index_col=0, header=0)
+multiplier_model_z = pd.read_csv("multiplier_model_z.tsv", sep="\t", index_col=0, header=0)
+
+
+# In[7]:
+
+
+multiplier_model_summary.head()
+
+
+# In[8]:
+
+
+# Only select LVs that are signficantly associated with some pathways or gene set (i.e. FDR < 0.05)
+multiplier_model_z_processed = process.process_multiplier_model_z(multiplier_model_z, multiplier_model_summary)
+
+
+# In[9]:
+
+
+multiplier_model_z_processed.head()
+
+
+# ## Quick looks at the data
+
+# In[10]:
+
+
+# Get a rough sense for how many genes contribute to a given LV
+# (i.e. how many genes have a value > 0 for a given LV)
+(multiplier_model_z > 0).sum()
+
+
+# In[11]:
+
+
+# One off just to get a sense for how many genes are being compared
+# Filter genes to only use those shared between our analysis and multiplier
+# Check overlap between multiplier genes and our genes
+multiplier_genes = list(multiplier_model_z.index)
+our_genes = list(pd.read_csv(ls_data_files[0], sep="\t", index_col=0, header=0).index)
+shared_genes = set(our_genes).intersection(multiplier_genes)
+
+print(len(our_genes))
+print(len(shared_genes))
+
+
+# ## Gene coverage of LV
+
+# In[12]:
+
+
+generic_cov_ls = []
+specific_cov_ls = []
+
+for ifile in range(len(ls_data_files)):
+    generic_genes = ls_genes_out[ifile][0]
+    specific_genes = ls_genes_out[ifile][1]
+    
+    # Only include those genes that are in multiplier otherwise will get NAs
+    generic_genes_processed, specific_genes_processed = process.process_generic_specific_gene_lists(
+        generic_genes, 
+        specific_genes, 
+        multiplier_model_z_processed
+    )
+    print(len(generic_genes_processed), len(specific_genes_processed))
+    
+    generic_cov_i, specific_cov_i = process.get_LV_coverage(
+        generic_genes_processed,
+        specific_genes_processed,
+        multiplier_model_z_processed
+    )
+    
+    generic_cov_ls += list(generic_cov_i)
+    specific_cov_ls += list(specific_cov_i)
+    
+from matplotlib_venn import venn2
+venn2([set(generic_cov_ls), set(specific_cov_ls)],
+      set_labels=("Generic LVs", "Specific LVs")
+     )
+
```

Comment:
So does the output here mean there are only 8 specific genes that are in the MultiPLIER results? And ~6000 generic genes? Are you at all surprised by these numbers (or am I interpreting them wrong)?

Reply:
The number of specific and generic genes are reported per simulation run. There are 4 runs each generating 25 simulated experiments and calculating z-score statistics based on these 25 experiments. Over the 4 runs half the time 0 specific genes were found and the other times there was 1 or 6 specific genes found. I was surprised by the number of specific genes being so few but this might be a result of the threshold I am using for the z-score cutoff and something that can be adjusted in the future.